src/rules-generator.ts

```diff
@@ -227,7 +225,7 @@ util.methods(Generator, {
   // Intermediate nodes can be "prop" or "$prop" values.
   createValidator: function(schemaName) {
     var schema = this.symbols.schema[schemaName];
-    var validator = {};
+    var validator: {$other?: any} = {}; // todo proper typing
```

Comment:
indeed, I somewhat myopically fixed a compile error, I am not sure what the type should really be, but its probably fundamental? You can type def a dictionary to give it a more meaningful name, without committing to a rigid structure (and then it makes refactoring over the validator type easy)
e.g. type Validator = {[key:name]:string}